src/components/Loadable/Loadable.spec.tsx

```diff
@@ -0,0 +1,148 @@
+    return element;
+  }
+  return element.querySelector(`[data-hook='linear-progressbar-tooltip']`);
+};
+
+describe('Loadable with sync loader', () => {
+  const createDriver = createDriverFactory(
+    loadableDriverFactory({
+      childFactory: tooltipDriverFactory,
+      childSelector,
+    }),
+  );
+
+  it('should load modules initially', () => {
+    const fallBackElement = <span data-hook="error-icon">Hey!</span>;
+    const wrapper = createDriver(
+      <LoadableTooltip
+        loader={() => require('../Tooltip')}
+        defaultComponent={fallBackElement}
+        componentKey="Tooltip"
+        shouldLoadComponent
+      >
+        {Tooltip => {
+          return (
+            <Tooltip
+              data-hook="linear-progressbar-tooltip"
+              placement="top"
+              content="kek"
+            >
+              {fallBackElement}
+            </Tooltip>
+          );
+        }}
+      </LoadableTooltip>,
+    );
+    expect(wrapper.isLoaded()).toBe(true);
+  });
+
+  it('should load modules after shouldLoadComponent changed', () => {
+    const tooltipSelector = '[data-hook="tooltip-child"]';
+    const wrapper = mount(
+      <LoadableTooltip
+        loader={() => require('../Tooltip')}
+        defaultComponent={<span data-hook="default-component">Hey!</span>}
+        componentKey="Tooltip"
+        shouldLoadComponent={false}
+      >
+        {Tooltip => {
+          return (
+            <Tooltip
+              data-hook="linear-progressbar-tooltip"
+              placement="top"
+              content="kek"
+            >
+              {<span data-hook="tooltip-child">Hey!</span>}
+            </Tooltip>
+          );
+        }}
+      </LoadableTooltip>,
+    );
+    expect(wrapper.find(tooltipSelector).exists()).toBe(false);
+    wrapper.setProps({ shouldLoadComponent: true });
+    setTimeout(() => {
+      expect(wrapper.find(tooltipSelector).exists()).toBe(true);
+    });
+  });
+});
+
+describe('Loadable with async loader', () => {
+  const createDriver = createDriverFactory(
+    loadableDriverFactory({
+      childFactory: tooltipDriverFactory,
+      childSelector,
+    }),
+  );
+
+  it('should load modules initially', async () => {
+    const fallBackElement = <span data-hook="error-icon">Hey!</span>;
+    const wrapper = createDriver(
+      <div>
+        <LoadableTooltip
+          loader={() => import('../Tooltip')}
+          defaultComponent={fallBackElement}
+          componentKey="Tooltip"
+          shouldLoadComponent
+        >
+          {Tooltip => {
+            return (
+              <Tooltip
+                data-hook="linear-progressbar-tooltip"
+                placement="top"
+                content="kek"
+              >
+                {fallBackElement}
+              </Tooltip>
+            );
+          }}
+        </LoadableTooltip>
+        ,
+      </div>,
+    );
+    expect(wrapper.isLoaded()).toBe(false);
```

Comment:
maybe it's enough for the driver just to return the rendered child, and then verify that the "linear-progressbar-tooltip" `data-hook` is on the element

Reply:
Here the problem is in not in abstraction, but in situation where we are receiving driver based on `element`. It's working ok when the it's wrapper stays the same, but not working when wrapper is being replaced with another one.
`Loader` is returning `defaultContent` or `loadingComponent` when the content wasn't loaded, but then it's returning the content from render props. So the shape is changing, which is not matching driver factory approach.